examples/quantum1d/7.xy-finiteT/main.jl

```diff
@@ -0,0 +1,471 @@
+              xlabel="β", ylabel="ΔF(β)", title="Free energy error")
+    plot(p1, p2)
+end
+
+md"""
+### MPO multiplication approach (linear)
+
+While the Taylor series approach is useful, we can only push that so far, since we are always expanding around $\beta = 0$.
+However, inspired by the trick we used to improve the results, we can use MPO multiplication techniques to compute partition functions at larger $\beta$.
+In particular, we can implement the following algorithm to scan over a linear range of $\beta$ values.
+
+```math
+\begin{align}
+Z(2\beta) &= Z(\beta) \cdot Z(\beta) \\
+Z(3\beta) &= Z(\beta) \cdot Z(\beta) \cdot Z(\beta) = Z(\beta) \cdot Z(2\beta) \\
+\vdots &= \vdots
+\end{align}
+```
+
+Multiplying two MPOs exactly would lead to an exponential growth in bond dimensions, but we can make use of standard MPS techniques to keep the bond dimensions under control.
+To achieve this, we can reinterpret the density matrix as an MPS with two physical indices.
+Then, we have some control over the approximations we make by tuning the maximal bond dimension.
+
+!!! warning
+    Using MPS techniques to approximate the multiplication of density matrices does not necessarily inherit all of the nice properties of approximating MPS.
+    In particular, the truncation of the MPO is now happening in the Frobenius norm, rather than the operator norm.
+    While for small truncations this might still work, this is not guaranteed to be the case for larger truncations.
+    As a result, the truncated object might not be positive semidefinite, spoiling its interpretation as a density matrix.
+"""
+
+Z_mpo_mul = zeros(length(βs))
+D_max = 64
+
+## first iteration: start from high order Taylor expansion
+ρ₀ = make_time_mpo(H, im * βs[2] / 2, TaylorCluster(; N=3))
+Z_mpo_mul[1] = Z_taylor[1]
+Z_mpo_mul[2] = real(dot(ρ₀, ρ₀))^(1 / N)
+
+## subsequent iterations: multiply by ρ₀
+ρ_mps = convert(FiniteMPS, ρ₀)
+for i in 3:length(βs)
+    global ρ_mps
+    @info "Computing β = $(βs[i])"
+    ρ_mps, = approximate(ρ_mps, (ρ₀, ρ_mps),
+                         DMRG2(; trscheme=truncdim(D_max), maxiter=10))
+    Z_mpo_mul[i] = real(dot(ρ_mps, ρ_mps))^(1 / N)
+end
+F_mpo_mul = -(1 ./ βs) .* log.(Z_mpo_mul)
+
+p_mpo_mul_diff = let
+    labels = reshape(map(expansion_orders[2:end]) do N
+                         return "Taylor N=$N"
+                     end, 1, :)
+    p1 = plot(βs, abs.(real.(Z_taylor2) .- partition_function.(βs, J, N));
+              label=labels, title="Partition function error",
+              xlabel="β", ylabel="ΔZ(β)")
+    plot!(p1, βs, abs.(real.(Z_mpo_mul) .- partition_function.(βs, J, N));
+          label="MPO multiplication")
+    p2 = plot(βs, abs.(real.(F_taylor2) .- free_energy.(βs, J, N)); label=labels,
+              xlabel="β", ylabel="ΔF(β)", title="Free energy error")
+    plot!(p2, βs, abs.(real.(F_mpo_mul) .- free_energy.(βs, J, N));
+          label="MPO multiplication")
+    plot(p1, p2)
```

Comment:
The legend in the second plot is positioned quite unfortunately. It would be better to move this to the northwest corner.